### DISCUSSION TITLE
How to run GitHub  Actions Workflow only for new tags

### DISCUSSION THREAD

**Discussion by brpaz** [community/community#25302] [Actions]
<p>Hello.</p>
<p>I have a Workflow that I only want to run when a new tag is pushed to the repository.</p>
<p>I have set the following “on” condition:</p>
<pre><code>on:
  create:
    tags:
</code></pre>
<p>But my workflow is being executed also for branches.</p>
<p>What I am missing?</p>
<p>Thank you.</p>

**Comment by ProvoK** [comment]
<p>+1</p>
<p>I’ve the same issue. It seems the tag filtering is not considered.</p>
<pre><code>on:
  create:
    tags:
      - v*
</code></pre>

**Comment by rokcarl** [comment]
<p>When I try to use on: I get a syntax error, e.g.</p>
<pre><code>workflow "MyWorkflow" {
  on:
    create:
      tags:
        - v*
  resolves = ["MyAction"]
}
</code></pre>
<p>gives me </p>
<pre><code>unable to parse
Line 2: illegal char
</code></pre>
<p>This is in .github/main.workflow, right?</p>

**Comment by rokcarl** [comment]
<p>Ah so GitHub have changed the syntax to YML, I’ve used their migration tool to get it “working”. Well, for me this means that using the syntax above means that for every tag push, I get a failed build and a message saying <em>No event triggers defined in on</em>.</p>

**Comment by rokcarl** [comment]
<p>Hmmm, that’s super weird, somehow I got it working with the exact code from <a class="mention" href="/provok">@provok</a> .</p>
<p>At first it wasn’t working for me, then I tried the exact code from GitHub’s <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#on" rel="nofollow noopener">workflow on</a> syntax and specified an exact tag v1. This worked, then I changed back to <a class="mention" href="/provok">@provok</a>’s code and it started working. Not sure what it was.</p>
<p>My steps now are: change some code, commit, push code. Change some new code, commit again and push. Then tag when I’m happy, push the tag. The latest commit will get the code build, but the first one will not, as expected.</p>

**Comment by glambert** [comment]
<p>Perhaps its because “create” also triggers when a branch is created, so it will run the filter on the first commit, then not until you actually create and push a new tag.</p>
<p><a class="mention" href="/brokge">@brokge</a> can you confirm that when you created the branch the workflow did not get triggered by the “create” action?</p>

**Comment by brpaz** [comment]
<p>It seems now that I cant even run workflows for tags:</p>
<p>I have this:</p>
<pre><code>on:
  push:
    tags:
       - '*'
</code></pre>
<p>But the workflow is not running at all when a new tag is pushed</p>

**Comment by fregante** [comment]
<p>None of this is working.</p>
<p>This is triggered even when branches are created. The <code>tags</code> part is just ignored:</p>
<pre><code>on:
  create:
    tags:
      - *
</code></pre>
<p>This will never be triggered:</p>
<pre><code>on:
  push:
    tags:
      - *
</code></pre>
<p>This will never be triggered either:</p>
<pre><code>on:
  push:
    branches:
      - 'refs/tags/*'
</code></pre>
<p>I’m creating tags with GitHub’s own “Create release” button.</p>
<p>Can anyone document this feature? </p>

**Comment by ethomson** [comment, marked as answer]
<p>In your second example, you need to quote the asterisk, that’s a YAML syntax requirement.  This should work:</p>
<pre><code>on:
  push:
    tags:
    - '*'
</code></pre>

**Comment by artalat** [comment]
<p>Does the same apply to branches as well? Because this doesn’t work for me:</p>
<pre><code>on:
  push:
    branches:
      - '*'
</code></pre>

**Comment by rectalogic** [comment]
<p>It does apply to branches, but if your branch might include / then you need to match on that too (for as many levels as you may have - */*, */*/*, */*/*/* etc.) e.g.:</p>
<pre><code>on:
  push:
    branches:
      - '*'
- '*/*'
</code></pre>

**Comment by brpaz** [comment]
<p>The proposed solutuon doesnt seem to work for me.</p>
<p>:</p>
<pre><code>on:
  push:
    tags:
    - '*'
</code></pre>

**Comment by ibakshay** [comment]
<p>The proposed solution doesn’t seem to work for me too :confused: . </p>

**Comment by ibakshay** [comment]
<p>I am sorry. It is indeed working for me. Thank you </p>

**Comment by p3lim** [comment]
<p><a class="mention" href="/artalat">@artalat</a> Use two asterisks, that’ll match sub-paths:</p>
<pre><code>on:
  push:
    branches:
      - '**'
</code></pre>

**Comment by jonfriesen** [comment]
<p>Hi <a class="mention" href="/ethomson">@ethomson</a> this solution appears to trigger when pushing to a new branch.</p>
<p>eg. <code>git push --set-upstream origin test</code> will cause it to trigger. Subsequent pushes will not cause it to trigger. Is this intended? </p>
<p>UPDATE:</p>
<p>This problem appears to have be fixed on the GitHub side?</p>

**Comment by oj22** [comment]
<p>where this needs to be updated.</p>
<p>I need to create trigger only when the tag is pushed/created in Github.any suggestion</p>

**Comment by SergeyLadutko** [comment]
<p>The assembly by the tag does not work, then ci starts when creating the branch</p>

**Comment by ob1dev** [comment]
<p>Hi <a href="/t5/user/viewprofilepage/user-id/47958">@ethomson</a> </p>
<p>What’s the recomended way to run a step (or even a job) based on a Git tag?</p>
<p>The esiest way I found out is as follows:</p>
<pre><code>steps:
-name: Check out from Git
 uses: actions/checkout@v1

-name: Check if Git tag exists
 run: echo "::set-env name=HEAD_TAG::$(git tag --points-at HEAD)"

-name: Runs only if Git tag exists
 run: echo"This runs because Git tag points at HEAD ..."
 if: env.HEAD_TAG != '' 
</code></pre>
<p>Would be handy if I could do like this:</p>
<pre><code>steps:
-name: Check out from Git
 uses: actions/checkout@v1

- name: Runs only if Git tag exists
    run: echo "This runs because Git tag points at HEAD ..."
    if:github.tag != '' 
</code></pre>
<p>Best Wishes,</p>
<ul>
<li>Oleg</li>
</ul>

**Reply by tomasfejfar** [reply]
> Hi [@ethomson](/t5/user/viewprofilepage/user-id/47958) 
> The esiest way I found out is as follows:
> 
> ```
> steps:
> -name: Check out from Git
>  uses: actions/checkout@v1
> -name: Check if Git tag exists
> run: echo "::set-env name=HEAD_TAG::$(git tag --points-at HEAD)"
> ```

Beware that there is an edge case when the tag action runs unexpectedly for a branch push for example, but there is a tag on the commit where the branch tip is.

**Comment by arunsathiya** [comment]
Is there a way to run the workflow only when tags are created in upstream repository? I can not seeing anything in that regard [in the docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore).

Or, is that supposed to be handled with an external action instead?
